posix/unix.c

```diff
@@ -866,17 +866,39 @@ int unix_shutdown(unsigned socket, int how)
 }
 
 
-/* TODO: support buffer reallocation */
+/* TODO: copy data from old buffer */
 static int unix_bufferSetSize(unixsock_t *s, int sz)
 {
-	if (sz < US_MIN_BUFFER_SIZE || sz > US_MAX_BUFFER_SIZE)
-		return -EINVAL;
+	void *v[2] = { NULL, NULL };
 
-	if (s->buffer.data != NULL)
+	if (sz < US_MIN_BUFFER_SIZE || sz > US_MAX_BUFFER_SIZE) {
 		return -EINVAL;
+	}
+
+	proc_lockSet(&s->lock);
+
+	if (s->buffer.data != NULL) {
+		proc_lockClear(&s->lock);
+
+		v[0] = vm_kmalloc(sz);
+		if (v[0] == NULL) {
+			return -ENOMEM;
+		}
+
+		proc_lockSet(&s->lock);
+
```

Comment:
I think we should either check here again if s->buffer.data is still != NULL, or don't let go of the lock :( Why we don't allocate a buffer if s->buffer data is NULL anyway? This would've make things simpler

Reply:
I can remove `proc_lockClear` if you are OK with `vm_kmalloc` under lock. Access to socket cbuffer (including checking s->buffer.data) should always be done under lock.

Reply:
Ok, it's a simpler solution, I'm ok with that

Reply:
The current one or the one without `proc_lockClear`?

Reply:
Current one is potentialy dangerous, so it's better to remove unlock/lock